VulkanLaunchpad.cpp

```diff
@@ -699,12 +700,34 @@ VkDeviceMemory vklAllocateHostCoherentMemoryForGivenRequirements(VkDeviceSize bu
 	}
 }
 
+VkDeviceMemory vklAllocateDeviceMemoryForGivenRequirements(VkDeviceSize bufferSize, VkMemoryRequirements memoryRequirements)
+{
+	const auto memoryAllocInfo = vklCreateMemoryAllocateInfo(bufferSize, memoryRequirements, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
+
+	// Allocate:
+	VkDeviceMemory memory;
+	VkResult returnCode = vkAllocateMemory(static_cast<VkDevice>(mDevice), &memoryAllocInfo, NULL, &memory);
+	if (returnCode == VK_SUCCESS) {
+		return memory;
+	}
+	else {
+		VKL_EXIT_WITH_ERROR(std::string("Error allocating memory of size [") + std::to_string(bufferSize) + "] and requirements[" + std::to_string(memoryRequirements.alignment) + ", " + std::to_string(memoryRequirements.memoryTypeBits) + ", " + std::to_string(memoryRequirements.size) + "]\n    Error Code: " + to_string(returnCode));
+	}
+}
+
 vk::UniqueDeviceMemory vklAllocateHostCoherentMemoryForGivenRequirements(vk::DeviceSize bufferSize, vk::MemoryRequirements memoryRequirements) {
-  const auto memoryAllocInfo = vklCreateMemoryAllocateInfo(bufferSize, memoryRequirements);
+  const auto memoryAllocInfo = vklCreateMemoryAllocateInfo(bufferSize, memoryRequirements, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
   auto allocatedMemory = mDevice.allocateMemoryUnique(memoryAllocInfo, nullptr, mDispatchLoader);
   return allocatedMemory;
 }
 
+vk::UniqueDeviceMemory vklAllocateDeviceMemoryForGivenRequirements(vk::DeviceSize bufferSize, vk::MemoryRequirements memoryRequirements) {
+	auto memoryAllocInfo = vklCreateMemoryAllocateInfo(bufferSize, memoryRequirements, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
+	auto allocatedMemory = mDevice.allocateMemoryUnique(memoryAllocInfo, nullptr, mDispatchLoader);
+	return allocatedMemory;
+}
+
+
```

Comment:
Same as above: 

Let's avoid code duplication and combine
- `vk::UniqueDeviceMemory vklAllocateHostCoherentMemoryForGivenRequirements(vk::DeviceSize bufferSize, vk::MemoryRequirements memoryRequirements)` and
- `vk::UniqueDeviceMemory vklAllocateDeviceMemoryForGivenRequirements(vk::DeviceSize bufferSize, vk::MemoryRequirements memoryRequirements)` 

into:
- `vk::UniqueDeviceMemory vklAllocateDeviceMemoryForGivenRequirements(vk::DeviceSize bufferSize, vk::MemoryRequirements memoryRequirements, vk::MemoryPropertyFlags flags)` !